src/components/App.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from 'react';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import './App.scss';
+import UserProfile from './UserProfile';
+import LightricksLogo from '../assets/lightricksLogo.svg';
+import { fetchUsers } from '../services/userService';
+import { Post, User } from '../types/User';
+import SearchContainer from './SearchContainer';
+
+const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [signedInUser, setSignedInUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const loadUsers = async () => {
+      try {
+        const data = await fetchUsers();
+        data.forEach(user => {
+          const posts: Post[] = [];
+          for (let i = 0; i < 3; i++) {
+            posts.push({
+              id: i,
+              image: `https://picsum.photos/200/200?random=${Math.floor(Math.random() * 1000)}`,
+              comments: Array(3)
+                .fill(null)
+                .map(() => ({
+                  body: 'This is a comment',
+                  userId: Math.floor(Math.random() * data.length) + 1,
+                })),
+            });
+          }
+          user.posts = posts;
+        });
+        setUsers(data);
+        setLoading(false);
+        setSignedInUser(data[Math.floor(Math.random() * data.length)]);
+      } catch (err) {
+        setError(err instanceof Error ? err.message : 'An error occurred');
+        setLoading(false);
+      }
+    };
+
+    loadUsers();
+  }, []);
+
+  const Layout = React.memo(({ children }: { children: React.ReactNode }) => (
+    <div className="layout">
+      <header className="main-header">
+        <img src={LightricksLogo} alt="Lightricks" className="Lightricks-icon" />
+        <img src={signedInUser?.image} alt="current user" className="current-user-image" />
+      </header>
+      <div className="app">{children}</div>
```

Comment:
i'd suggest using scss modules:
https://github.com/css-modules/css-modules 
https://css-tricks.com/css-modules-part-1-need/
(this is css, not scss, but the idea is the same - you name your file `App.module.scss` and then import it like `import styles from "App.module.scss"`, and reference classname as styles.className)